apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx

```diff
@@ -0,0 +1,183 @@
+import { sentTransactionStatusChanged } from '@/features/sentTransactions/sentTransactionsActions'
+import { selectSentTransactionByHash } from '@/features/sentTransactions/sentTransactionsSelectors'
+import SnackbarBox from '@/features/snackbar/SnackbarBox'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import { SentTransaction } from '@/types/transactions'
+
+const SentTransactionSnackbarPopup = memo(({ txHash }: { txHash: string }) => {
+  const dispatch = useAppDispatch()
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+  const [hide, setHide] = useState(false)
+
+  const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
+
+  useEffect(() => {
+    if (!tx) return
+
+    dispatch(sentTransactionStatusChanged({ hash: tx.hash, status: isConfirmedTx(tx) ? 'confirmed' : 'mempooled' }))
+  }, [dispatch, tx])
+
+  useEffect(() => {
+    let timer: ReturnType<typeof setTimeout>
+
+    if (sentTx?.status === 'confirmed') {
+      timer = setTimeout(() => setHide(true), 5000)
+    }
+
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+  }, [sentTx?.status])
+
+  if (!sentTx || hide) return null
+
+  return (
+    <SentTransactionSnackbarPopupStyled {...fadeInBottom} {...fadeOut} className="info">
+      <Columns>
+        <Rows>
+          <Message status={sentTx.status} />
+
+          <HashEllipsedStyled hash={txHash} tooltipText={t('Copy hash')} showSnackbarOnCopied={false} />
+
+          <Progress status={sentTx.status} />
+        </Rows>
+        <ButtonsRow>
+          {sentTx.status !== 'sent' && <DetailsButton hash={txHash} />}
+
+          <Button aria-label={t('Close')} squared role="secondary" onClick={() => setHide(true)} borderless transparent>
+            <X />
+          </Button>
+        </ButtonsRow>
+      </Columns>
+    </SentTransactionSnackbarPopupStyled>
+  )
+})
+
+export default SentTransactionSnackbarPopup
+
+const SentTransactionSnackbarPopupStyled = styled(SnackbarBox)`
+  min-width: 400px;
+`
+
+const Message = ({ status }: Pick<SentTransaction, 'status'>) => {
+  const { t } = useTranslation()
+
+  if (status === 'sent') return `${t('Transaction was sent...')} 💸`
+
+  if (status === 'mempooled') return `${t('Transaction is about to be included in the blockchain...')} ⏳⛓️`
+
+  if (status === 'confirmed') return `${t('Transaction is now part of the blockchain')} 🎉`
+}
+
+const Progress = ({ status }: Pick<SentTransaction, 'status'>) => {
+  const [progress, setProgress] = useState(0)
+
+  useEffect(() => {
+    if (status === 'sent') {
+      setProgress(0.1)
+    } else if (status === 'mempooled') {
+      setProgress((prevValue) => (prevValue < 0.25 ? 0.25 : prevValue))
+    } else if (status === 'confirmed') {
+      setProgress(1)
+    }
+  }, [status])
+
+  useInterval(() => setProgress((prevValue) => prevValue + 0.015), 1000, status === 'confirmed' || progress > 0.9)
```

Comment:
UI comment: after testing it, I think I'd rather have a spinner which transforms in a nice "check" animation when the TX is confirmed, rather than a fake progress bar.

Reply:
Would you like to work on it, with your amazing animation skills? :D IT'd be nice if the loading animation can morph into a check!

Reply:
I wanted to somehow visualize the progress of moving from "sent" to "mempooled" to "confirmed". So I went with `10%` progress increase for "sent", `25%` for "mempooled" with the message "your tx is about to be included in the blockchain" and then gradually increase by `1.5%` per second to a max of `90%` where we stop and wait before the tx comes "confirmed" and we move the progress to `100%`.

Reply:
We could also do:
- `10%` sent
- `50%` mempooled
- `100%` confirmed

with no fake progress. But I thought since our blocktime is 16s, a fake progress would not be THAT fake in most of the cases.

Reply:
Ok! Let's keep the progress steps then. I'll redesign this using a circle progress bar to be more subtle ;)